core/src/main/scala/org/apache/spark/serializer/JavaSerializer.scala

```diff
@@ -92,9 +106,14 @@ private object JavaDeserializationStream {
 }
 
 private[spark] class JavaSerializerInstance(
-    counterReset: Int, extraDebugInfo: Boolean, defaultClassLoader: ClassLoader)
+    counterReset: Int,
+    extraDebugInfo: Boolean,
+    defaultClassLoader: ClassLoader,
+    useCache: Boolean)
   extends SerializerInstance {
 
+  lazy val resolvedClassesCache = new ConcurrentHashMap[(String, ClassLoader), Class[_]]()
```

Comment:
Rather than be lazy, maybe just only instantiate the map if the cache is used?

Reply:
Also, how big might this map get? I suppose its lifetime is limited, and won't cache just massive numbers of classes, hm.

Reply:
updated to Option. In the RPC module, the `JavaSerializerInstance` lives along with `RpcEnv`, but the size is limited by the RPC types.